code/modules/research/xenobiology/crossbreeding/warping.dm

```diff
@@ -0,0 +1,848 @@
+		/obj/item/flashlight/flashdark,
+		/mob/living/simple_animal/slime/rainbow,
+		/obj/item/storage/belt/sabre,
+		/obj/item/drone_shell,
+		/obj/item/sharpener,
+		/mob/living/simple_animal/hostile/cat_butcherer
+	)
+
+	var/static/list/rare_items = list(
+		/obj/effect/mob_spawn/human/syndicate/battlecruiser/captain,
+		/obj/structure/spawner/skeleton,
+		/obj/effect/spawner/lootdrop/armory_contraband,
+		/obj/effect/spawner/lootdrop/teratoma/major
+	)
+
+
+/obj/effect/warped_rune/goldspace/do_effect(mob/user)
+	var/price = 0
+	var/list/valuable_items = list()
+	for(var/obj/item/I in rune_turf)
+		var/datum/export_report/ex = export_item_and_contents(I, dry_run=TRUE)
+		for(var/x in ex.total_amount)
+			if(ex.total_value[x])
+				price += ex.total_value[x]
+				valuable_items |= I
+
+	if(price >= target_value)
+		deleteme = TRUE
+		var/num = rand(1,100)
+		var/path
+
+		switch(num)
+			if(1 to 80)
+				path = pick(common_items)
+			if(80 to 95)
+				path = pick(uncommon_items)
+			else
+				path = pick(rare_items)
+
+		var/atom/movable/A = new path(rune_turf)
+		for(var/obj/item/I in valuable_items)
+			I.loc = null
+			qdel(I)
+		to_chat(user, "<spawn class='notice'>[src] shines and [A] appears before you.</span>")
+	else
+		to_chat(user, "<span class='warning'>The sacrifice is insufficient.</span>")
+	. = ..()
+
+//oil
+/obj/item/slimecross/warping/oil
+	colour = "oil"
+	runepath = /obj/effect/warped_rune/oilspace
+	effect_desc = "Draw a rune that can explode whoever steps on it."
+
+/obj/effect/warped_rune/oilspace
+	icon_state = "rune_oil"
+	desc = "This is basically a mine."
+	deleteme = FALSE
+
+/obj/effect/warped_rune/oilspace/Crossed(atom/movable/AM, oldloc)
+	if(iscarbon(AM))
+		var/mob/living/carbon/C = AM
+		var/amt = rand(4,12)
+		C.reagents.add_reagent(/datum/reagent/water, amt)
+		C.reagents.add_reagent(/datum/reagent/potassium, amt)
+		activated = TRUE
+	. = ..()
+
+/obj/item/slimecross/warping/black
+	colour = "black"
+	runepath = /obj/effect/warped_rune/blackspace
+	effect_desc = "Draw a rune that can transmute a corpse into a shade."
+
+/obj/effect/warped_rune/blackspace
+	icon_state = "rune_black"
+	desc = "Souls are like any other material, you just have to find the right place to manufacture them."
+
+/obj/effect/warped_rune/blackspace/do_effect(mob/user)
+	for(var/mob/living/carbon/human/host in rune_turf)
+		if(host.key) //checks if the ghost and brain's there
+			to_chat(user, "<span class='warning'>This body can't be transmuted by the rune in this state!</span>")
+			return
+
+		to_chat(user, "<span class='warning'>The rune is trying to repair [host.name]'s soul!</span>")
+		var/list/candidates = pollCandidatesForMob("Do you want to replace the soul of [host.name]?", ROLE_SENTIENCE, null, ROLE_SENTIENCE, 50, host, POLL_IGNORE_SHADE)//todo: fix desc
+
+		else if(length(candidates)) //check if anyone wanted to play as the dead person and check if no one's in control of the body one last time.
```

Comment:
ah shit my bad, the suggested `else if` didn't account for the lines between it 😅

Reply:
```suggestion
		if(length(candidates) && !host.key) //check if anyone wanted to play as the dead person and check if no one's in control of the body one last time.
```

Reply:
There's probably a better way to do it but this works fine